frontend/src/metabase-types/api/insight.ts

```diff
@@ -21,7 +21,7 @@ export type InsightExpression =
 
 export interface Insight {
   col: string;
-  unit: DatetimeUnit;
+  unit: DateTimeAbsoluteUnit;
```

Comment:
It's safe to narrow the type because in `insights.clj` we see that the `unit` field will only ever be one of the absolute date units

```clojure

(def ^:private unit->duration
  {:minute  (/ 1 24 60)
   :hour    (/ 24)
   :day     1
   :week    7
   :month   30.5
   :quarter (* 30.4 3)
   :year    365.1})
```